chaos-days/blog/2023-12-19-Dynamic-Scaling-with-Dataloss/index.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: posts
+title:  "Dynamic Scaling with Dataloss"
+date:   2023-12-19
+categories: 
+  - chaos_experiment 
+  - bpmn
+tags:
+  - availability
+authors: ole
+---
+
+# Chaos Day Summary
+
+We continue our [previous expirements](../2023-12-18-Dynamically-scaling-brokers/index.md) with dynamically scaling by now also testing whether
+the cluster survives dataloss during the process.
+
+One goal is to verify that we haven't accidentally introduced a single point of failure in the cluster.
+Another is to ensure that data loss does not corrupt the cluster topology.
+
+**TL;DR;**
+Even with dataloss, the scaling completes successfully and with the expected results.
+We found that during scaling, a single broker of the previous cluster configuration can become a single point of failure by preventing a partition from electing a leader.
+This is not exactly a bug, but something that we want to improve.
+
+<!--truncate-->
+
+## Dataloss on the Coordinator
+
+Zeebe uses Broker 0 as the coordinator for changes to the cluster topology.
+While changes can only be initiated by the coordinator, losing the coordinator and it's data should not prevent the scaling operation from completing.
+When the coordinator restarts without any data, it should be able to recover the cluster topology as well as the partition data from the remaining brokers.
+
+To test this, we use the `zbchaos dataloss delete` and `zeebe dataloss recover` commands.
+After deleting, the broker will not restart directly, instead waiting for the `zbchaos dataloss recover` command to be executed.
+
+Shortly after triggering a scale up with `zbchaos cluster scale --brokers 6`, we trigger dataloss on the coordinator with `zbchaos broker dataloss delete --nodeId 0`.
+After observing the system for a while, we then restore the coordinator with `zbchaos dataloss recover --nodeId 0`.
+
+### Expected
+
+The scaling operation eventually completes with the expected result of 6 brokers and 6 partitions, evenly distributed.
+The coordinator recovers after dataloss and eventually receives the cluster topology from the remaining brokers.
+The scaling operation should make progress while the coordinator is down.
+
+### Actual
+
+After starting the operation with `zbchaos cluster scale --brokers 6` we see that the operation has started.
+We then trigger dataloss on the coordinator with `zbchaos broker dataloss delete --nodeId 0`.
```

Comment:
🔧 How do you see this? Metrics/ logs, something you want to share here?

Reply:
I didn't take screenshots for that part unfortunately. I think it's okay though - the interesting bit is what happens after.